flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -197,13 +196,19 @@ private static LocalResource registerLocalResource(
     }
 
     public static void setTokensFor(
-            ContainerLaunchContext amContainer, List<Path> paths, Configuration conf)
+            org.apache.flink.configuration.Configuration flinkConf,
+            ContainerLaunchContext amContainer,
+            Configuration hadoopConf)
             throws IOException {
         Credentials credentials = new Credentials();
-        // for HDFS
-        TokenCache.obtainTokensForNamenodes(credentials, paths.toArray(new Path[0]), conf);
-        // for HBase
-        obtainTokenForHBase(credentials, conf);
+
+        // obtain tokens from HadoopDelegationTokenProviders
```

Comment:
Need to be compatible with https://issues.apache.org/jira/browse/FLINK-21700#

Reply:
Merged with FLINK-21700